components/script/script_runtime.rs

```diff
@@ -125,6 +136,73 @@ unsafe extern "C" fn enqueue_job(cx: *mut JSContext,
     }), false)
 }
 
+#[allow(unsafe_code)]
+/// https://html.spec.whatwg.org/multipage/#the-hostpromiserejectiontracker-implementation
+unsafe extern "C" fn promise_rejection_tracker(
+    cx: *mut JSContext,
+    promise: HandleObject,
+    state: PromiseRejectionHandlingState,
+    _data: *mut c_void
+) {
+    let global = GlobalScope::from_context(cx);
+
+    wrap_panic(AssertUnwindSafe(|| {
+        match state {
+            PromiseRejectionHandlingState::Unhandled => {
+                global.add_uncaught_rejection(promise);
+            },
+            PromiseRejectionHandlingState::Handled => {
+                global.add_consumed_rejection(promise);
+            }
+        };
+    }), ());
+}
+
+#[allow(unsafe_code, unrooted_must_root)]
+/// https://html.spec.whatwg.org/multipage/#notify-about-rejected-promises
+pub unsafe fn notify_about_rejected_promises(global: DomRoot<GlobalScope>) {
```

Comment:
Let's take `&GlobalScope` as an argument instead. And instead of making this function unsafe, we should use unsafe blocks in the implementation.